### PR TITLE
[feat](stats) Return job id for async analyze stmt

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

1. Return job id for async analyze stmt
2. Sync analysis jobs doesn't persist to analysis_jobs anymore

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: regression-test/suites/statistics/analyze_test.groovy [hunk 1]
```diff
@@ -232,4 +232,66 @@ suite("analyze_test") {
...
+    sql """
+        insert into __internal_schema.analysis_jobs values(788943185,-1,'internal','default_cluster:analyze_test_db_1',
+        'analyze_test_tbl_1', 'analyze_test_col3',-1 ,'MANUAL','sFULL','',0,'PENDING', 'ONCE');
+    """
+
+    sql """
+        DROP EXPIRED STATS
+    """
+
+//    Exception e = null;
```
Comment (morrySnow) [inline regression-test/suites/statistics/analyze_test.groovy:244]: useless code?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java [hunk 2]
```diff
@@ -87,6 +88,13 @@ public class StatisticsRepository {
     private static final String DROP_TABLE_STATISTICS_TEMPLATE = "DELETE FROM " + FeConstants.INTERNAL_DB_NAME
             + "." + "${tblName}" + " WHERE ${condition}";
 
+    private static final String FIND_EXPIRED_JOBS = "SELECT job_id FROM "
+            + FULL_QUALIFIED_ANALYSIS_JOB_TABLE_NAME
+            + " WHERE task_id = -1 AND ${now} - last_exec_time_in_ms  > "
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java:93]: what's mean about `task_id = -1`

File: regression-test/suites/statistics/analyze_test.groovy (hunk 1 above)
Comment (Kikyou1997) [reply regression-test/suites/statistics/analyze_test.groovy:244]: Just comment it temporarily

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java (hunk 2 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java:93]: We only need job_id here, `task_id=-1` is used to filter task item in the `analysis_jobs` table

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Pull request merged**